tests/test_window_streams.py

```diff
@@ -3,160 +3,182 @@
+    for window_ms, overlap_ms, scenario in valid_configs:
+        config = WindowConfig(window_ms=window_ms, overlap_ms=overlap_ms)
+        assert config.window_ms == window_ms, scenario
+        assert config.overlap_ms == overlap_ms, scenario
+
+    # Given: Invalid window configurations with expected error messages
+    invalid_configs = [
+        (100.0, 100.0, "Overlap must be less than window size"),
+        (100.0, 150.0, "Overlap must be less than window size"),
+        (0.0, 0.0, "Window size must be positive"),
+        (-100.0, 50.0, "Window size must be positive"),
+    ]
+
+    # When/Then: Creating invalid configurations raises ValueError with expected message
+    for window_ms, overlap_ms, expected_msg in invalid_configs:
+        with pytest.raises(ValueError, match=expected_msg):
+            WindowConfig(window_ms=window_ms, overlap_ms=overlap_ms)
+
+
+def test_sample_conversion():
+    """Test conversion from milliseconds to samples"""
+    # Given: A window configuration
+    config = WindowConfig(window_ms=500.0, overlap_ms=250.0)
+    sample_rate = 250  # Hz
+
+    # When: Converting to samples
+    window_samples, overlap_samples = config.to_samples(sample_rate)
 
-    # Invalid configurations
-    with pytest.raises(ValueError):
-        WindowConfig(window_ms=100.0, overlap_ms=100.0)  # Equal window and overlap
-    with pytest.raises(ValueError):
-        WindowConfig(window_ms=100.0, overlap_ms=150.0)  # Overlap > window
+    # Then: Sample counts should match expected values
+    assert window_samples == 125, "500ms at 250Hz should be 125 samples"
+    assert overlap_samples == 62, "250ms at 250Hz should be 62 samples"
 
 
-def test_basic_window_shape():
-    """Test that windows have the expected shape"""
-    config = WindowConfig(window_ms=500.0, overlap_ms=250.0)  # Known good values
+def test_window_generation():
+    """Test basic window generation functionality"""
+    # Given: A standard window configuration
+    config = WindowConfig(window_ms=500.0, overlap_ms=250.0)
 
     with board_stream() as board:
+        # When: Generating windows
+        generator = stream_windows(board, config)
+        window = next(generator)
+
+        # Then: Window shape should match configuration
         sample_rate = board.get_sampling_rate(board_id=board.board_id)
         expected_samples = int(sample_rate * config.window_ms / 1000)
         eeg_channels = board.get_eeg_channels(board_id=board.board_id)
 
-        generator = stream_windows(board, config)
-        window = next(generator)
-
-        assert isinstance(window, np.ndarray)
-        assert window.shape == (len(eeg_channels), expected_samples)
+        assert isinstance(window, np.ndarray), "Window should be numpy array"
+        assert window.shape == (len(eeg_channels), expected_samples), "Window shape incorrect"
+        assert not np.any(np.isnan(window)), "Window should not contain NaN values"
 
 
 def test_window_overlap():
```

Comment:
Can we add a test where the window_ms, frequency of sampling, and overlap_ms share no common factors?

Reply:
We certainly can.